Add an initial GitHub Actions config to run Safari stable

gsnedders opened the pull request:
We'll likely want to refactor this into a reusable workflow, especially for Safari Technology Preview, but potentially for any other browsers we want to run on GitHub Actions.

That said, let's start with the simple singular case and make sure this works before adding more complexity.

gsnedders committed:
Add an initial GitHub Actions config to run Safari stable

We'll likely want to refactor this into a reusable workflow,
especially for Safari Technology Preview, but potentially for any
other browsers we want to run on GitHub Actions.

However, land this without this refactoring, as the important thing is
to land this (and migrate away from Azure Pipelines) in the short term
due to impending capacity constraints.

This will likely fail in the wpt_fyi_notify.yml workflow until
https://github.com/web-platform-tests/wpt.fyi/commit/594077b9c49e026214df4ad5670983260fd98082
is deployed to wpt.fyi.

foolip commented on .github/workflows/safari_stable.yml:
```diff
@@ -0,0 +1,87 @@
+      - triggers/safari_stable
+
+env:
+  safaridriver_diagnose: false
```
Is this here as a clue that it can be set to true for debugging? If so, can you add a comment?

gsnedders replied:
Oh, I should just copy/paste the comment from the prior Azure Pipelines pipeline.

foolip commented on .github/workflows/safari_stable.yml:
```diff
@@ -0,0 +1,87 @@
+      - name: Enable safaridriver
+        run: |-
+          set -eux -o pipefail
+          export SYSTEM_VERSION_COMPAT=0
```
I guess this is harmless, but wasn't this a workaround for something quite long ago that's no longer needed?

gsnedders replied:
Oh, for `safaridriver` this absolutely shouldn't be needed.

For actually running tests, I believe this depends on what SDK the binaries we're running (primarily the Python interpreter) are built against — if they're built against a pre-macOS 11 SDK, then without this macOS will claim to be 10.16.

Given this setup is using the Xcode-provided Python, we can probably entirely get rid of this.

foolip reviewed (commented):
Did it work? Is there an artifacts upload somewhere to inspect?

jgraham reviewed (approved):
Assuming this works it LGTM. For preference I'd have all the steps that are just running shell commands as a single shell script that can be run outside of CI rather than inlining them into the workflow file, but it's not a blocker.

gsnedders commented:
> Did it work? Is there an artifacts upload somewhere to inspect?

https://github.com/gsnedders/web-platform-tests/actions/runs/9980202391

> Assuming this works it LGTM. For preference I'd have all the steps that are just running shell commands as a single shell script that can be run outside of CI rather than inlining them into the workflow file, but it's not a blocker.

This largely just matches what Azure Pipelines does, albeit with everything in the workflow instead of split into many actions. This is also arguably overly simplistic, because we really want the workflow to be reusable for multiple browsers, which means we need steps to be conditional — but it seemed easiest to inline everything and iterate from here.

jgraham commented:
OK, well happy for this to be merged as-is.

gsnedders committed:
Add workflow for Safari Technology Preview

This duplicates most of safari_stable.yml, which shows that we really
do need to refactor most of this into a reusable workflow, but the
short-term win of continuing to have results for Safari outweighs
purity here.
